src/mqt/yaqs/simulator.py

```
@@ -371,11 +383,20 @@ def _call_backend(backend: Callable[[Any], TRes], arg: Any, n_threads: int = 1)
         - If ``threadpoolctl`` is not available, falls back to direct call.
         - If enforcing thread limits fails, falls back silently to direct call.
     """
+    # Numba threading must be set BEFORE the threadpoolctl context limits backend threads,
+    try:
+        import numba  # noqa: PLC0415
```

Ruff (0.15.1)

[warning] Unused `noqa` directive (non-enabled: `PLC0415`)

Remove unused `noqa` directive

(RUF100)